### PR TITLE
Fix image script

### PR DESCRIPTION
**Pull request by delta2323**
I fix the evaluation script of networks for image recognition, so that it align with Torch implementation of DeepMark. I also change 
- Measure whole training time, do not split them into each iteration, nor into forward, backward and upward.
- Add dry run option.
- Feed ndarray's directly without wrapping them with chainer Variable.
- Use special classifier for inception v3
- Use cleargrad.

### CODE REVIEW

File: deepmark_chainer/net/inception_v3.py [hunk 1]
```diff
@@ -235,4 +236,21 @@ def classifier(x, train):
         y = classifier(x, train)
         self.y = y
         self.y_aux = y_aux
-        return y
+        return y, y_aux
+
+
+class Classifier(link.Chain):
```
Comment (bkvogel) [inline deepmark_chainer/net/inception_v3.py]: This uses the same name as the Classifier in Chainer. Consider renaming to something more specific for clarity.

File: evaluate/train_image.py [hunk 2]
```diff
@@ -1,115 +1,115 @@
...
     in_size = 224
 else:
     raise ValueError('Invalid architector:{}'.format(args.predictor))
-model = L.Classifier(predictor)
+
 
 if args.gpu >= 0:
     cuda.get_device(args.gpu).use()
     model.to_gpu()
 optimizer = O.SGD()
```
Comment (bkvogel) [inline evaluate/train_image.py]: The Torch7 deepmark benchmark uses learning rate = 0.0001. Since we are operating on random data anyway, it does not make a practical difference, but might still be good idea to keep the code and model hyperparameters as similar as possible.

File: evaluate/train_image.py [hunk 3]
```diff
@@ -1,115 +1,115 @@
...
-        update_time += update_time_one
 
-forward_time /= args.iteration
-backward_time /= args.iteration
-update_time /= args.iteration
+total_time = t.total_time()
+averate_gime = total_time / args.iteration
 
-print('Mean\t{}\t{}\t{}'.format(forward_time, backward_time, update_time))
+print('Total Time\t{}'.format(total_time))
```
Comment (bkvogel) [inline evaluate/train_image.py]: Consider printing the output in a similar format as the corresponding deepmark benchmark. That is, print the average time per iteration and also print the number of examples/second.

**Review by bkvogel — CHANGES_REQUESTED** [review]
I think it looks good and the code might be okay as-is, but please consider my suggested changes. Thanks.

File: evaluate/train_image.py (hunk 2 above)
Comment (delta2323) [reply evaluate/train_image.py]: Fixed

File: deepmark_chainer/net/inception_v3.py (hunk 1 above)
Comment (delta2323) [reply deepmark_chainer/net/inception_v3.py]: Fixed

File: evaluate/train_image.py (hunk 3 above)
Comment (delta2323) [reply evaluate/train_image.py]: Fixed

**Comment by delta2323** [comment]
@bkvogel Thank you for your comments. I updated the code according to your suggestion.

**Review by bkvogel — APPROVED** [review]
LGTM.
